odilia/src/main.rs

```diff
@@ -80,6 +85,42 @@ async fn main() -> eyre::Result<()> {
 	//initialize a task tracker, which will allow us to wait for all tasks to finish
 	let tracker = TaskTracker::new();
 
+	tracing::debug!("Reading configuration");
```

Comment:
Could you move the creation of the application config to a new function in `config`? Seems more fitting. In fact, you could even implement this as `ApplicationConfigBuilder::default()` or something like that, as it constructs a config from various steps.

Reply:
I don't think that's a very good idea. As I see it, the configuration is supposed to be plane objects, models as in an mvc pattern for example, essentially just data.

Adding methods to that objects would imply that the configuration knows how to build itself, aka has additional responsabilities. This is contrary to the principles I go by writing code, because essentially config is indeed just a typed data store. That's why it should live in state, with no mutability possibilities, passing around bits of it to the various services needing those.

That's why I think this should be done in main, if not the main function, at least the main module. You see, the configuration should be initialised first, because it initialises everything else, including logging. State::new already needs logging, for various things, especially because it then goes on and initialises lots of the core components of the screenreader, which in turn need logging, and orphan spans will be weird in the visualisers or even if we inspect the logs by hand.